src/app/features/fss-search/fss-search.component.ts

```diff
@@ -247,4 +253,42 @@ export class FssSearchComponent implements OnInit {
     this.eventAdvancedSearchTokenRefresh.next();
   }
 
+  transformSearchAttributesToFilter(attributeSearchResult: any) {
+    let configAttributes: any[] = [];
+    this.filterGroups = [];
+
+    configAttributes = AppConfigService.settings["fssConfig"].batchAttributes ;
+      
+      if(configAttributes.length > 0 && attributeSearchResult.length > 0)
+      {
+       let result = configAttributes.every((item: any) => {
+        var attribute = attributeSearchResult.filter((function(data: { key: any; }) {
```

Comment:
use const instead of var

if you are just looking for the first attribute that has a certain `key` then the `.find()` method is probably more appropriate than the `.filter()` method.

i believe somethng like 

```
const attribute = attributeSearchResults.find(searchResult => searchResult.key === item);

if (attribute) {
  // do stuff
}
```

if the compiler complains about accessing ` searchResult.key` you can use ` searchResult["key"]` instead but i think it should be fine.

Reply:
updated as suggested. Thanks